src/posts/posts.service.ts

```diff
@@ -1,26 +1,80 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, NotFoundException } from '@nestjs/common';
 import { CreatePostDto } from './dto/create-post.dto';
 import { UpdatePostDto } from './dto/update-post.dto';
+import { Post } from './entities/post.entity';
 
 @Injectable()
 export class PostsService {
+  private posts: Post[] = [
+    {
+      id: 1,
+      userId: 1,
+      title: '1 - 1',
+      content: 'user 1, id 1',
+    },
+    {
+      id: 2,
+      userId: 1,
+      title: '1 - 2',
+      content: 'user 1, id 2',
+    },
+    {
+      id: 3,
+      userId: 2,
+      title: '2 - 3',
+      content: 'user 2, id 3',
+    },
+  ];
   create(createPostDto: CreatePostDto) {
-    return 'This action adds a new post';
+    // new ID generation
+    const newId = this.posts.length > 0 
+      ? Math.max(...this.posts.map(p => p.id)) + 1 
+      : 1;
+    
+    const newPost: Post = {
+      id: newId,
+      ...createPostDto,
+    };
+    
+    this.posts.push(newPost);
+    return newPost;
   }
 
   findAll() {
-    return `This action returns all posts`;
+    return this.posts;
+  }
+
+  findByUserId(userId: number) {
+    return this.posts.filter(post => post.userId === userId);
   }
 
   findOne(id: number) {
-    return `This action returns a #${id} post`;
+     const post = this.posts.find(post => post.id === id);
+    if (!post) {
+      throw new NotFoundException(`Post with ID ${id} not found`);
+    }
+    return post;
   }
 
   update(id: number, updatePostDto: UpdatePostDto) {
-    return `This action updates a #${id} post`;
+    const postIndex = this.posts.findIndex(post => post.id === id);
```

Comment:
You can just use method "find". If then, you don't have to always do like "this.posts[postIndex]".

Reply:
Btw, findIndex method also work!